backend/stylists/app/Http/Controllers/TipController.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+
+use App\Models\Enums\EntityType;
+use App\Models\Enums\EntityTypeName;
+use App\Models\Enums\RecommendationType;
+use App\Models\Lookups\Lookup;
+use App\Models\Lookups\AppSections;
+use App\Models\Lookups\Status;
+use App\Tip;
+
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redirect;
+use Validator;
+
+class TipController extends Controller
+{
+    
+    protected $filter_ids   = ['age_group_id', 'gender_id','created_by', 'body_type_id', 'budget_id', 'occasion_id', 'status_id'];
+    protected $filters      = ['age_groups', 'genders','createdBy', 'body_types', 'budgets', 'occasions', 'statuses'];
+
+    
+    public function index(Request $request, $action, $id = null, $action_id = null)
+    {
+        $method = strtolower($request->method()) . strtoupper(substr($action, 0, 1)) . substr($action, 1);
+        
+        if($id) {
+            
+            $this->resource_id = $id;
+            
+        }
+        
+        if($action_id) {
+            
+            $this->action_resource_id = $action_id;
+            
+        }
+
+        return $this->$method($request);
+    }
+
+    public function getCreate(Request $request)
+    {
+            
+            return view('tip.create');
+    }
+    
+    public function postCreate(Request $request)
+    {
+
+        $tip = new Tip();
+        
+        $tip->name          = isset($request->name) && $request->name != '' ? $request->name : '';
+        $tip->description   = isset($request->description) && $request->description != '' ? $request->description : '';
+        $tip->image         = isset($request->image) && $request->image != '' ? $request->image : '';
+        $tip->image_url     = isset($request->image_url) && $request->image_url != '' ? $request->image_url : '';
+        $tip->video_url     = isset($request->video_url) && $request->video_url != '' ? $request->video_url : '';
+        $tip->external_url  = isset($request->external_url) && $request->external_url != '' ? $request->external_url : '';
+        $tip->budget_id     = isset($request->budget_id) && $request->budget_id != '' ? $request->budget_id : '';
+        $tip->age_group_id  = isset($request->age_group_id) && $request->age_group_id != '' ? $request->age_group_id : '';
+        $tip->body_type_id  = isset($request->body_type_id) && $request->body_type_id != '' ? $request->body_type_id : '';
+        $tip->occasion_id   = isset($request->occasion_id) && $request->occasion_id != '' ? $request->occasion_id : '';
+        $tip->gender_id     = isset($request->gender_id) && $request->gender_id != '' ? $request->gender_id : '';
+        $tip->created_by    = $request->user()->id != '' ? $request->user()->id : '';
```

Comment:
put validation rules for all the fields name, image , image_url, video_url, external_url, gender_id.